### PR TITLE
Added custom context for popout & throw an exception incase of auth chat

### PR DESCRIPTION
**Pull request by dishakhattri**
Exception :
image

image

### CODE REVIEW

File: chat-widget/src/components/livechatwidget/common/startChat.ts [hunk 1]
```diff
@@ -219,11 +217,21 @@ const canConnectToExistingChat = async (props: ILiveChatWidgetProps, chatSDK: an
...
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const setCustomContextParams = (props: ILiveChatWidgetProps, chatSDK: any) => {
+const setCustomContextParams = (chatSDK: any) => {
     // Add custom context only for unauthenticated chat
     const persistedState = getStateFromCache(chatSDK?.omnichannelConfig?.orgId, chatSDK?.omnichannelConfig?.widgetId, props?.controlProps?.widgetInstanceId ?? "");
 
-    if (!props.chatConfig?.LiveChatConfigAuthSettings && !isUndefinedOrEmpty(persistedState?.domainStates?.customContext)) {
+    if (!isUndefinedOrEmpty(persistedState?.domainStates?.customContext)) {
+        if(persistedState?.domainStates.liveChatConfig?.LiveChatConfigAuthSettings) {
+            const errorMessage = "Use of custom context with authenticated chat is deprecated. The chat would not go through.";
```
Comment (elopezanaya) [inline chat-widget/src/components/livechatwidget/common/startChat.ts:226]: why not use a constant value as the other values for the error object?

File: chat-widget/src/components/livechatwidget/common/startChat.ts [hunk 2]
```diff
@@ -219,11 +217,21 @@ const canConnectToExistingChat = async (props: ILiveChatWidgetProps, chatSDK: an
 };
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const setCustomContextParams = (props: ILiveChatWidgetProps, chatSDK: any) => {
+const setCustomContextParams = (chatSDK: any) => {
     // Add custom context only for unauthenticated chat
     const persistedState = getStateFromCache(chatSDK?.omnichannelConfig?.orgId, chatSDK?.omnichannelConfig?.widgetId, props?.controlProps?.widgetInstanceId ?? "");
```
Comment (elopezanaya) [inline chat-widget/src/components/livechatwidget/common/startChat.ts:222]: if there any point to get persistedState here, when we may not use it?, I think a better place is after LOC 224,

**Review by elopezanaya — CHANGES_REQUESTED** [review]
couple of comments, and validations are failing

File: chat-widget/src/components/livechatwidget/common/startChat.ts (hunk 1 above)
Comment (dishakhattri) [reply chat-widget/src/components/livechatwidget/common/startChat.ts:230]: As talked over call, will fix as per another task for all hardcoded messges.

**Review by dishakhattri — COMMENTED** [review]
Resolved comments

**Review by elopezanaya — APPROVED** [review]

**Pull request merged**